src/Database/Table/Selection.php

```diff
@@ -265,7 +265,9 @@ public function fetchAll()
 	 */
 	public function fetchAssoc($path)
 	{
-		return Nette\Utils\Arrays::associate($this->fetchAll(), $path);
+		return Nette\Utils\Arrays::associate(array_map(function(IRow $row) {
+			return iterator_to_array($row);
+		}, $this->fetchAll()), $path);
```

Comment:
~~~php
Nette\Utils\Arrays::associate(array_map('iterator_to_array', $this->fetchAll()))
~~~